sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/parquet/ParquetRelation.scala

```diff
@@ -208,7 +208,15 @@ private[sql] class ParquetRelation(
   // Parquet data source always uses Catalyst internal representations.
   override val needConversion: Boolean = false
 
-  override def sizeInBytes: Long = metadataCache.dataStatuses.map(_.getLen).sum
+  override def sizeInBytes: Long =
+    if (shouldMergeSchemas && mergeRespectSummaries) {
+      // If we are going to merge schema and this relation is configured to
+      // respect summaries (i.e., skip part-files), we will assume that the size of
+      // this relation is large and can't be broadcasted.
```

Comment:
This assumption doesn't seem right to me... It's perfectly OK for a small Parquet dataset to require schema merging and to be configured to respect summaries.